Flagsmith.FlagsmithClient/AnalyticsProcessor.cs

```diff
@@ -79,9 +94,53 @@ public async Task Flush()
         /// <returns></returns>
         public async Task TrackFeature(string featureName)
         {
-            AnalyticsData[featureName] = AnalyticsData.TryGetValue(featureName, out int value) ? value + 1 : 1;
-            if ((DateTime.Now - _LastFlushed).Seconds > _FlushIntervalSeconds)
+            string threadId = Thread.CurrentThread.ManagedThreadId.ToString();
+            // Get thread-specific count dictionary
+            Dictionary<string, int> threadAnalyticsData;
+            if (!AnalyticsDataThreads.TryGetValue(threadId, out threadAnalyticsData))
+            {
+                threadAnalyticsData = new Dictionary<string, int>();
+                AnalyticsDataThreads[threadId] = threadAnalyticsData;
+            }
+
+            // Increment local count
+            int count;
+            if (!threadAnalyticsData.TryGetValue(featureName, out count))
+            {
+                count = 0;
+            }
+            count++;
+            threadAnalyticsData[featureName] = count;
+
+            int _LastFlushedInterval = (DateTime.Now - _LastFlushed).Seconds;
+
+            if (_LastFlushedInterval > _FlushIntervalSeconds)
                 await Flush();
         }
+
+        /// <summary>
+        /// Get aggregated analytics data.
+        /// </summary>
+        /// <returns>Dictionary of feature name and usage count</returns>
+        public Dictionary<string, int> GetAggregatedAnalytics()
```

Comment:
This needs some better documentation, or maybe it's sufficient (and better) to document all this behaviour on the class itself. It's quite complicated with the separate threads, etc.

Reply:
Thanks, done